### PR TITLE
metrics: add percentage of CPU and memory usage gauges (#8539)

### LINKED ISSUE #8539: Collect CPU metrics for TiDB Server
## Feature Request

**Is your feature request related to a problem? Please describe:**

At present, we can not know the CPU usage of the `tidb-server` process in the monitor.

**Describe the feature you'd like:**

As the title said, a simple demo is:
```go
package main

import (
    "fmt"
    "os"
    "sync"
    "time"

    "github.com/struCoder/pidusage"
)

func printStat() {
    sysInfo, err := pidusage.GetStat(os.Getpid())
    if err != nil {
        fmt.Printf("%v\n", err)
    } else {
        fmt.Printf("CPU: %v, MEM: %v\n", sysInfo.CPU, sysInfo.Memory)
    }
}

func useCPU(wg *sync.WaitGroup) int {
    defer wg.Done()

    sum := 1
    for i := 0; i < 1000000000; i++ {
        if i%997 == 0 {
            sum += i
        } else {
            sum -= i
        }
    }
    return sum
}

func main() {
    wg := sync.WaitGroup{}
    wg.Add(8)
    for i := 0; i < 8; i++ {
        go useCPU(&wg)
    }

    time.Sleep(2 * time.Second)
    printStat()

    wg.Wait()
}
```

**Describe alternatives you've considered:**

Maybe there are other libs that can obtain the CPU usage of a process in golang.

**Teachability, Documentation, Adoption, Migration Strategy:**

No

### PR DESCRIPTION
**Pull request by ziyi-yan**
### What problem does this PR solve? 
fixes #8539 

### What is changed and how it works?
Two gauges are added for the percentages of CPU and memory usage. 

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-ansible` repository
 - Need to be included in the release note

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8580)

### CODE REVIEW

**Comment by ziyi-yan** [comment]
PTAL @zz-jason @shenli

File: tidb-server/main.go [hunk 1]
```diff
@@ -479,13 +480,23 @@ func setupMetrics() {
...
 			metrics.KeepAliveCounter.Inc()
+			updateCPUUsageMetrics()
 		}
 	}
 	go systimemon.StartMonitor(time.Now, systimeErrHandler, sucessCallBack)
 
 	pushMetric(cfg.Status.MetricsAddr, time.Duration(cfg.Status.MetricsInterval)*time.Second)
 }
 
+func updateCPUUsageMetrics() {
```
Comment (zz-jason) [inline tidb-server/main.go]: We should setup a background goroutine to update the CPU usage metrics periodically?

File: metrics/server.go [hunk 2]
```diff
@@ -152,6 +152,22 @@ var (
...
+
+	CPUUsagePercentageGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "tidb",
+			Subsystem: "server",
+			Name:      "cpu_usage",
+			Help:      "Percentage CPU usage.",
+		})
+
+	MemUsagePercentageGauge = prometheus.NewGauge(
```
Comment (zz-jason) [inline metrics/server.go]: We already have the metrics about memory usage, this metrics is duplicated.

File: tidb-server/main.go (hunk 1 above)
Comment (eurekaka) [reply tidb-server/main.go]: Seems `go systimemon.StartMonitor(time.Now, systimeErrHandler, sucessCallBack)` achieves this?

Comment (zz-jason) [reply tidb-server/main.go]: Sorry, I missed this function `go systimemon.StartMonitor(time.Now, systimeErrHandler, sucessCallBack)`. I think it's no need to create another goroutine. 😄

**Comment by zz-jason** [comment]
@eurekaka @tiancaiamao @jackysp PTAL

**Pull request merged**